lib/client/Client.js

```diff
@@ -120,29 +118,53 @@ export default class Client {
         clientName: Constants.CLIENT_NAME,
         clientVersion: Constants.VERSION
       })
-      .then(data => {
-        authContinuation = new AuthContinuation(data);
+      .then(data => this._authenticateResponse(data, continuationCallback));
+  }
 
-        return authContinuation;
-      })
-      .then(authContinuation => continuationCallback(authContinuation))
-      .then(continueData => {
-        if (authContinuation.methods.indexOf(continueData.method) < 0) {
-          throw new Error('The "' + continueData.method + '" authentication protocol is not supported by the server.');
-        }
-        let param = {
-          token: authContinuation.continuationToken,
-          method: continueData.method
-        };
+  /**
+   * Sub-routine handling JMAP server responses on authentication requests
+   *
+   * Depending on the server's response, this either (recursively) executes the second
+   * authentication step by calling the provided continuationCallback or resolves on
+   * successfully completed authentication.
+   *
+   * @param data {Object} The JMAP response data
+   * @param continuationCallback {Function} The callback function initially passed to the authenticate() method
+   *
+   * @return {Promise} A {@link Promise} that will eventually be resovled with a {@link AuthAccess} object
+   */
+  _authenticateResponse(data, continuationCallback) {
+    var authContinuation;
```

Comment:
I'm afraid that's not possible. `data` can either be an `AuthContinuation ` or an `AuthAccess` model. Creating an `AuthContinuation` instance here already might thrown an exception if it's the latter case.